include/RAJA/policy/openmp_target/forall.hpp

```diff
@@ -60,9 +60,10 @@ RAJA_INLINE void forall_impl(const omp_target_parallel_for_exec<ThreadsPerTeam>&
   }
 
 // thread_limit(tperteam) unused due to XL seg fault (when tperteam != distance)
+  auto i = distance_it;
```

Comment:
What's this line for?

Reply:
The old version used `for(decltype(distance_it) i = 0 ; ...)` to declare the iterator variable, for some reason clang was generating the incorrect type there and causing signed/unsigned warnings on the loop comparison.  This declaration is a shortcut to getting a decayed type declaration to use for that which gets around the problem.

Reply:
Ah, that makes sense - thanks for the explanation.

Reply:
Thanks for asking actually, I'll add a comment because it's really only there to work around what I'd call a compiler bug.